Make serverless compute the default for the deployed recon job (#2565)
databrickslabs/remorph: docs/lakebridge/docs/reconcile/index.mdx

The User configuring reconcile must have permission to:
- Create Data Warehouses
- `USE CONNECTION` on the source connection
- `USE CATALOG` and `CREATE SCHEMA` on the target catalog
- `CREATE VOLUME` if using a pre-existing schema

> - Create Compute Clusters — only when running the job on a classic cluster via `job_overrides` (see below)

Reviewer: ```suggestion
```

Author: Done — removed the bullet in 2b3a7acf.

### Compute selection

**The deployed reconcile job runs on serverless compute by default.** The lakebridge wheel is

> supplied through a job-level serverless environment — no cluster is created or managed for you.

Reviewer: ```suggestion
**The deployed reconcile job runs on serverless compute by default.** no cluster is created or managed for you.
```

Author: Done in 2b3a7acf — took the suggestion (capitalized the "No").

To run the job on a classic cluster instead, create the cluster yourself and point
`job_overrides` at it in the reconcile config (`<USER_WORKSPACE_HOME>/.lakebridge/reconcile.yml`):

```yaml
job_overrides:

>   existing_cluster_id: "0714-000000-abcdefgh"

Reviewer: we also need empty tags for correct deserialization

Author: Done — added `tags: {}` to the example in 2b3a7acf.

```

:::note
A classic cluster must run **Databricks Runtime 17.3 or above** — the `remote_query()`

> table-valued function used to read from the source connection is not available on older runtimes.

Reviewer: ```suggestion
A classic cluster must run **Databricks Runtime 17.3 or above** that supports reading from Unity Catalog connections.
```

Author: Done in 2b3a7acf.
